src/crypto/vault.rs

```diff
@@ -0,0 +1,119 @@
+use crate::consts::base64::BASE64_ENGINE;
+use crate::crypto::{Crypto, Source};
+use crate::env::observability as logger;
+use crate::errors::{self, CryptoError, CustomResult, SwitchError};
+use base64::Engine;
+use error_stack::report;
+use futures::Future;
+use masking::{PeekInterface, Secret, StrongSecret};
+use serde::Deserialize;
+use std::pin::Pin;
+#[cfg(feature = "vault")]
+use vaultrs::client::{VaultClient, VaultClientSettingsBuilder};
+#[cfg(feature = "vault")]
+use vaultrs::{api, transit};
+#[derive(Debug, Deserialize, Clone)]
+pub struct VaultSettings {
+    url: String,
+    mount_point: String,
+    encryption_key: String,
+}
+
+pub struct Vault {
+    inner_client: VaultClient,
+    settings: VaultSettings,
+}
+
+pub fn init_vault(
+    settings: VaultSettings,
+    token: Secret<String>,
+) -> CustomResult<Vault, CryptoError> {
+    let client = VaultClient::new(
+        VaultClientSettingsBuilder::default()
+            .address(&settings.url)
+            .token(token.peek())
+            .build()
+            .switch()?,
+    )
+    .switch()?;
+    Ok(Vault {
+        inner_client: client,
+        settings: settings,
+    })
+}
+
+#[async_trait::async_trait]
+impl Crypto for Vault {
+    type DataReturn<'a> = Pin<
+        Box<
+            dyn Future<Output = CustomResult<StrongSecret<Vec<u8>>, errors::CryptoError>>
+                + 'a
+                + Send,
+        >,
+    >;
+
+    async fn generate_key(
+        &self,
+    ) -> CustomResult<(Source, StrongSecret<[u8; 32]>), errors::CryptoError> {
+        let response = transit::generate::random_bytes(
```

Comment:
Can you also add comments for the options? Like what is RandBytesSource::All and why did we choose it